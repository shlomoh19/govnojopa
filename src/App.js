import React, { lazy, Suspense, useCallback, useEffect, useState } from 'react';
import Header from "./components/header/Header";
import { Route } from "react-router-dom";
import Home from "./components/home/Home";
import Footer from "./components/footer/Footer";
import MobileMenu from './components/mobile-menu/Mobile-menu';
import Contacts from './components/contacts/Contacts';
import Spinner from './components/spinner/Spinner';
import BefreeSkeleton from './components/be-free/be-free-skeleton';
import CastingSkeleton from './components/casting/CastingSkeleton';
import './App.scss';
import CookiePopup from './components/cookie-popup/cookie-popup';
import { getCookiesByName, setAgreeWithCookies } from './utils/cookies';



const BefreeLazy = lazy(() => import('./components/be-free/Be-free'))
const CastingPageLazy = lazy(() => import('./components/casting/Casting-page'))

function App() {
    const [mobileMenu, setMobileMenu] = useState(false)
    const [done, setDone] = useState(false)
    const [isShowCookiesPopup, setIsShowCookiesPopup] = useState(false)

    useEffect(() => {
        const isAgree = !!getCookiesByName('agreeWithCookies')
        if (!isAgree) {
            setIsShowCookiesPopup(true)
        } else {
            setIsShowCookiesPopup(false)
        }
    }, [isShowCookiesPopup])

    const mobileMenuHandler = useCallback(() => setMobileMenu(!mobileMenu), [mobileMenu])

    const menuActive = {
        opacity: "1",
        zIndex: '5'
    }

    const menuUnActive = {
        opacity: '0',
        zIndex: '-1'
    }

    const setAgreeCookie = () => {
        setAgreeWithCookies()
        setIsShowCookiesPopup(false)
    }

    return (
        <Suspense fallback={<Spinner />}>
            <div className={mobileMenu ? "App Open-menu-body" : "App"}>
                <Header
                    mobileMenuHandler={mobileMenuHandler}
                    mobileMenu={mobileMenu}
                >
                    <MobileMenu mobileMenu={mobileMenu} mobileMenuHandler={mobileMenuHandler} />
                    <div className="content-wrapper">
                        <div onClick={mobileMenuHandler} style={mobileMenu ? menuActive : menuUnActive} className="Open-menu-content" />
                        <Route exact path="/" component={Home} />
                        <Route path="/befree/:id" render={() => {
                            return (
                                <Suspense fallback={<BefreeSkeleton />}>
                                    <BefreeLazy done={done} setDone={setDone} />
                                </Suspense>
                            )
                        }} />
                        <Route path="/contacts" render={() => <Contacts done={done} setDone={setDone} />} />

                        <Route path="/casting" render={() => {
                            return (
                                <Suspense fallback={<CastingSkeleton />}>
                                    <CastingPageLazy done={done} setDone={setDone} />
                                </Suspense>
                            )
                        }} />
                    </div>
                    <Footer />
                </Header>
            </div>
            {isShowCookiesPopup && <CookiePopup setCookiePopup={setAgreeCookie} />}
        </Suspense>
    );
}

export default App;
