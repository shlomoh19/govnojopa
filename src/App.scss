.Open-menu-content {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, .5);
  z-index: 5;
  transition: all .5s;
  opacity: 1;
}

.content-wrapper {
  width: 70%;
  margin: 0 auto;
  font-size: 20px;
}

@media (max-width: 1440px) {
  .content-wrapper {
    font-size: 16px;
  }
}

@media (max-width: 900px) {
  .content-wrapper {
    font-size: 14px;
  }
}
@media (max-width: 800px) {
  .content-wrapper {
    width: 90%;
  }
}
