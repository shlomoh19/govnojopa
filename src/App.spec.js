import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import App from './App';

describe('App', () => {
    it("should render App component", () => {
        const component = shallow(<App />);
        const wrapper = component.find(".content-wrapper");
        expect(wrapper.length).toBe(1);
    });

    it('renders correctly', () => {
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
