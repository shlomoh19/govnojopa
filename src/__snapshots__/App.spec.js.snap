// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Suspense
  fallback={<Spinner />}
>
  <div
    className="App"
  >
    <Header
      mobileMenu={false}
      mobileMenuHandler={[Function]}
    >
      <MobileMenu
        mobileMenu={false}
        mobileMenuHandler={[Function]}
      />
      <div
        className="content-wrapper"
      >
        <div
          className="Open-menu-content"
          onClick={[Function]}
          style={
            Object {
              "opacity": "0",
              "zIndex": "-1",
            }
          }
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          path="/befree/:id"
          render={[Function]}
        />
        <Route
          path="/contacts"
          render={[Function]}
        />
        <Route
          path="/casting"
          render={[Function]}
        />
      </div>
      <Suspense
        fallback={
          <div>
            Loading...
          </div>
        }
      >
        <lazy />
      </Suspense>
    </Header>
  </div>
</Suspense>
`;

exports[`App should render App snapshot 1`] = `ShallowWrapper {}`;
